components/script/dom/headers.rs

```diff
@@ -86,6 +86,7 @@ impl HeadersMethods for Headers {
         if let Some(v) = self.header_list.borrow().get_raw(&valid_name) {
             combined_value = v[0].clone();
             combined_value.push(b","[0]);
+            combined_value.push(b" "[0]);
```

Comment:
I wasn't aware that you were working with the raw bytes of the Header. In that case, it's likely more according to the spec to add a new 'line'. It seems that there is missing method to do this, `get_raw_mut`, but it would otherwise look like this:

```rust
match headers.get_raw_mut("foo") {
    Some(mut raw) => raw.push(value),
    None => headers.set_raw("foo", vec![value])
}
```